core/store/lib/search.ts

```diff
@@ -17,120 +19,41 @@ export function isOnline () {
-      console.error('Offline mode ' + err.message)
-    } else {
-      throw new Error(err)
-    }
-  })
-}
-/**
- * Helper function to handle ElasticSearch Results
- * @param {Object} resp result from ES call
- * @param {Int} start pagination data
- * @param {Int} size pagination data
- */
-function _handleEsResult (resp, start = 0, size = 50): ESResponse {
-  if (resp == null) {
-    if (isOnline()) {
-      throw new Error('Invalid ES result - null not exepcted')
-    } else {
-      return null // empty result but not exception raised - we're in offline mode
-    }
-  }
-  if (resp.hasOwnProperty('hits')) {
-    return {
-      items: map(resp.hits.hits, (hit) => {
-        return Object.assign(hit._source, { _score: hit._score, slug: (hit._source.hasOwnProperty('url_key') && rootStore.state.config.products.useMagentoUrlKeys) ? hit._source.url_key : (hit._source.hasOwnProperty('name') ? slugify(hit._source.name) + '-' + hit._source.id : '') }) // TODO: assign slugs server side
-      }), // TODO: add scoring information
-      total: resp.hits.total,
-      start: start,
-      perPage: size,
-      aggregations: resp.aggregations
-    }
-  } else {
-    if (resp.error) {
-      throw new Error(JSON.stringify(resp.error))
-    } else {
-      throw new Error('Unknown error with ES result in _handleEsResult')
-    }
-  }
-}
-
 /**
  * Search ElasticSearch catalog of products using simple text query
  * Use bodybuilder to build the query, aggregations etc: http://bodybuilder.js.org/
- * @param {Object} query elasticSearch request body
+ * @param {Object} query is the object of searchQuery class
  * @param {Int} start start index
  * @param {Int} size page size
  * @return {Promise}
  */
-export function quickSearchByQuery ({ query, start = 0, size = 50, entityType = 'product', sort = '', index = null, excludeFields = null, includeFields = null }): Promise<ESResponse> {
+export function quickSearchByQuery ({ query, start = 0, size = 50, entityType = 'product', sort = '', storeCode = null, excludeFields = null, includeFields = null }): Promise<Response> {
   if (size <= 0) size = 50
   if (start < 0) start = 0
 
   return new Promise((resolve, reject) => {
     const storeView = currentStoreView()
-    const esQuery: ESQuery = {
-      index: index || storeView.elasticsearch.index, // TODO: add grouped product and bundled product support
+    const Request: Request = {
+      store: storeCode || storeView.storeCode, // TODO: add grouped product and bundled product support
       type: entityType,
-      body: query,
+      searchQuery: query,
+      groupToken: null,
+      groupId: null,
       size: size,
       from: start,
       sort: sort
     }
 
-    if (excludeFields) esQuery._sourceExclude = excludeFields
-    if (includeFields) esQuery._sourceInclude = includeFields
+    if (excludeFields) Request._sourceExclude = excludeFields
+    if (includeFields) Request._sourceInclude = includeFields
```

Comment:
**This is important**

We must ensure that graphQL driver is taking care of `_sourceInclude` and `_sourceExclude` as they are key to optimize the bandwidth of data requests:

we’re running two queries in that case - here You have the result on Elastic (demo): (https://www.dropbox.com/s/acu4l0f868ovndf/Screenshot%202018-09-11%2017.31.23.png?dl=0) 

and in graphql - the result is exactly the same in size because _excludeFields parameter is not being used (https://www.dropbox.com/s/znwfqrbvfn8llas/Screenshot%202018-09-11%2017.30.41.png?dl=0

In first case you see the difference: 123 vs 320kb - in second case the excludeFields are ignored